src/modelframe.jl

```diff
@@ -172,19 +172,40 @@ function setcontrasts!(mf::ModelFrame, new_contrasts::Dict)
 end
 setcontrasts!(mf::ModelFrame; kwargs...) = setcontrasts!(mf, Dict(kwargs))
 
+convertinsupport(v::AbstractVector, d::Distribution) = Vector{partype(d)}(v)
+
+function convertinsupport(v::Union{AbstractCategoricalVector, AbstractNullableCategoricalVector},
+    d::Bernoulli)
+    vdrp = droplevels!(copy(v))
+    levs = CategoricalArrays.levels(vdrp)
+    llv = length(levs)
+    if llv < 2
+        zeros(vdrp, partype(d))
+    elseif llv == 2
+        partype(d)[get(x) == levs[2] for x in vdrp]
```

Comment:
`get` won't work for non-nullable arrays. If you want the most efficient code, that would be:
```julia
x = get(v.pool, levs[2])
partype(d)[ref == x for ref in v.refs]
```